rest/route/host.go

```diff
@@ -16,13 +16,106 @@ import (
 	"github.com/pkg/errors"
 )
 
+////////////////////////////////////////////////////////////////////////
+//
+// PATCH /rest/v2/hosts
+
+type hostsChangeStatusesHandler struct {
+	HostToStatus map[string]map[string]string
```

Comment:
This structure is a bit hard to reason about, at least for me. Would it be viable to define a type (even locally in this package,) to parse data into? rather than just relying on maps?